### PR TITLE
nie

### PR DESCRIPTION
**Pull request by Nieadni**
nie

### CODE REVIEW

File: src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
+package com.github.manasmods.tensura.gui.screen;
+
+import com.github.manasmods.manascore.client.gui.widget.ImagePredicateButton;
+import com.github.manasmods.tensura.Tensura;
+import com.github.manasmods.unordinary_basics.utils.Translation;
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java:5]: Why are u importing something from UB?

File: src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    private static final ResourceLocation ARROW_LEFT = new ResourceLocation(Tensura.MOD_ID, "textures/gui/manga/arrow_left.png");
+    private static final ResourceLocation ARROW_RIGHT = new ResourceLocation(Tensura.MOD_ID, "textures/gui/manga/arrow_right.png");
+
+    private int page = 1; //Store the current page in the screen
+    private static final int maxPage = 51; //store the max amount of pages in the screen
+    private ResourceLocation currentPageRL; //cache the current resource location
+
+    private ImagePredicateButton backButton;
+    private ImagePredicateButton forwardButton;
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java:35]: Make them final

File: src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+        this.currentPageRL = this.getCurrentPageRL();
+    }
+
+    @Override
+    protected void init() {
+        super.init();
+
+        this.backButton = new ImagePredicateButton(0, 0, 32, 16, ARROW_LEFT, this::OnPressBack, (pButton, pPoseStack, pMouseX, pMouseY) -> {}, () -> true);
+        this.forwardButton = new ImagePredicateButton(0, 0, 32, 16, ARROW_RIGHT, this::OnPressForward, (pButton, pPoseStack, pMouseX, pMouseY) -> {}, () -> true);
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java:48]: move this to the constructor and set the x and y position in the init method

File: src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        if(this.page > 1) {
+            this.page--;
+
+            this.currentPageRL = this.getCurrentPageRL();
+        }
+    }
+
+    private ResourceLocation getNextPageRL() {
+        return new ResourceLocation(Tensura.MOD_ID, String.format("textures/gui/manga/0001-%03d.png", page));
+    }
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java:68]: remove this method since getCurrentPageRL does the same

**Review by Charismara — CHANGES_REQUESTED** [review]

**Comment by Charismara** [comment]
please fix the build error!

File: src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java (hunk 1 above)
Comment (Nieadni) [reply src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java:5]: Was testing if the disc worked on the jukebox awhile back

File: src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java [hunk 5]
```diff
@@ -31,11 +21,13 @@ public class Chapter1Screen extends Screen {
...
-    private ImagePredicateButton backButton;
-    private ImagePredicateButton forwardButton;
+    private final ImagePredicateButton backButton;
+    private final ImagePredicateButton forwardButton;
 
-    public Chapter1Screen(Component pTitle) {
+    public Chapter1Screen(Component pTitle, ImagePredicateButton backButton, ImagePredicateButton forwardButton) {
         super(pTitle);
+        this.backButton = backButton;
+        this.forwardButton = forwardButton;
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/gui/screen/Chapter1Screen.java:30]: what is going on here?
Why are there buttons in the constuctor params?

File: src/main/java/com/github/manasmods/tensura/item/custom/Chapter1Item.java [hunk 6]
```diff
@@ -26,7 +26,7 @@ public InteractionResultHolder<ItemStack> use(Level pLevel, Player pPlayer, Inte
         pPlayer.awardStat(Stats.ITEM_USED.get(this));
 
         if (pLevel.isClientSide) {
-            Minecraft.getInstance().setScreen(new Chapter1Screen(new TextComponent("chapter1ItemGui")));
+            Minecraft.getInstance().setScreen(new Chapter1Screen(new TextComponent("chapter1ItemGui"), backButton, forwardButton));
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/tensura/item/custom/Chapter1Item.java:29]: what is going on here?
Why are there buttons in the constuctor params?

**Review by Charismara — CHANGES_REQUESTED** [review]

**Pull request merged**